ClientSupport/src/main/java/io/deephaven/clientsupport/gotorow/SeekRow.java

```diff
@@ -46,37 +49,38 @@ public SeekRow(long startingRow, String columnName, Object seekValue, boolean in
     }
 
     @Override
-    @ConcurrentMethod
     public Long apply(Table table) {
-        final int sortDirection = guessSorted(table);
-        final boolean isSorted = !contains && sortDirection != 0;
-
+        final Optional<SortingOrder> order = SortedColumnsAttribute.getOrderForColumn(table, columnName);
         final RowSet index = table.getRowSet();
-        long row;
-        if (isSorted) {
-            final Comparable value =
-                    (Comparable) table.getColumnSource(columnName).get((int) index.get((int) startingRow));
-            final int compareTo =
-                    sortDirection * nullSafeCompare(value, (Comparable) seekValue) * (isBackward ? -1 : 1);
-            final int start = isBackward ? (int) startingRow + 1 : (int) startingRow;
-            if (compareTo == 0) {
-                return startingRow;
-            } else if (compareTo < 0) {
-                // value is less than seek value
-                log.info().append("Value is before: ").append(nullSafeToString(value)).append(" < ")
-                        .append(nullSafeToString(seekValue)).endl();
-                row = maybeBinarySearch(table, index, sortDirection, start, (int) index.size() - 1);
+
+        if (!order.isEmpty()) {
+            if (isBackward) {
+                // check prev row
+                if (startingRow != 0) {
+                    final Comparable prevValue =
+                            (Comparable) table.getColumnSource(columnName).get((int) index.get((int) startingRow - 1));
+                    if (nullSafeCompare(prevValue, (Comparable) seekValue) == 0) {
+                        return startingRow - 1;
+                    }
+                }
+                // no values before, loop back around and find the last value
```

Comment:
Unless I'm misunderstanding, `startingRow` is not necessarily positioned on a match. It might be a random row the user has clicked.

Consequently, I don't think we've shown that there are "no values before". I think you are missing a binary search in the values before.

Same for the forward case.